### PR TITLE
Resolve the 7.0 upgrade checklist

### PR DESCRIPTION
**Pull request by begriffs**
A running PR to add commits to as we go through the checklist

### CODE REVIEW

**Comment by begriffs** [comment]
Fixes #435

**Comment by begriffs** [comment]
See commit messages for details about UDFs and GUCs omitted.

File: installation/requirements.rst [hunk 1]
```diff
@@ -3,7 +3,7 @@
 Requirements
 ############
 
-Citus works with modern 64-bit Linux and most Unix based operating systems. Citus 6.0 requires PostgreSQL 9.5 or later versions.
+Citus works with modern 64-bit Linux and most Unix based operating systems. Citus 6.x requires PostgreSQL 9.5 or later versions, and Citus 7.0 requires PostgreSQL 9.6 or later.
```
Comment (sumedhpathak) [inline installation/requirements.rst]: Can we just update this to Citus 7? (Instead of adding onto version 6).

File: reference/configuration.rst [hunk 2]
```diff
@@ -49,6 +49,16 @@ This value defaults to true, and is effective on the coordinator. In rare cases,
 complex upgrade processes may require setting this parameter to false, thus
 disabling the check.
 
+citus.log_distributed_deadlock_detection (boolean)
```
Comment (sumedhpathak) [inline reference/configuration.rst]: These look good, but I might still add `max_task_string_size`. We usually talk about the type of the param, min/max values etc, which might be useful.

File: reference/user_defined_functions.rst [hunk 3]
```diff
@@ -719,6 +718,35 @@ Example
   --------------
   73 MB
 
+citus.find_groupid_for_node
```
Comment (sumedhpathak) [inline reference/user_defined_functions.rst]: I don't think this is meant to be external. It also seems low-level. `master_add_secondary_node` probably should be added though. The function comment is here: https://github.com/citusdata/citus/blob/f1b51d7bbe4178e2cd5fb08959e7a4cd87bd6a3b/src/backend/distributed/utils/node_metadata.c#L174. The arguments are different from this and master_add_node. In this function, we also specify which is the primary node for this one.

File: cloud/mx.rst [hunk 4]
```diff
@@ -107,7 +107,7 @@ An important aspect to consider is that horizontally scaling out your processing
 Limitations Compared to Citus
 -----------------------------
 
-All Citus 6.2 features are supported in Citus MX with the following exceptions:
+All Citus 7.0 features are supported in Citus MX with the following exceptions:
```
Comment (sumedhpathak) [inline cloud/mx.rst]: This probably needs updating, but let's skip that for now.

File: conf.py [hunk 5]
```diff
@@ -57,9 +57,9 @@
 # built documents.
 #
 # The short X.Y version.
-version = '6.2'
+version = '7.0'
```
Comment (sumedhpathak) [inline conf.py]: This was merged as part of Burak's PR. Could you revert, or rebase with master?

File: installation/requirements.rst (hunk 1 above)
Comment (sumedhpathak) [reply installation/requirements.rst]: Ping on this one.

**Pull request merged**